### ISSUE TITLE
[Feature Request] PMD reports contain absolute paths

### ISSUE THREAD

**Issue by bh-sergio-clara** [forcedotcom/sfdx-scanner#1185]
**Describe the bug**
Even though PMD provides a mechanism to relativize paths (see `--relativize-paths-with` in [docs](https://docs.pmd-code.org/latest/pmd_userdocs_cli_reference.html)) however that's not being passed down by scanner which means that reports will come out referencing files with absolute paths.

We'd like to store the results with our code (to keep track of progress, calculate detas vs. some baseline, for example) and this makes it impossible if you have a team sharing the code or would cause problems if the project changes location.

**To Reproduce**
`sfdx scanner:run --engine=pmd --target=force-app/main/default/classes/SomeClass.cls  --pmdconfig=scanner/some-rule-set.xml --format=json`

```
[
  {
    "engine": "pmd-custom",
    "fileName": "/Users/MyUser/MyRepo/force-app/main/default/classes/BadClass.cls",
    "violations": [
      {
        "line": 211,
        "column": 5,
        "endLine": 211,
        "endColumn": 36,
        "severity": 3,
        "ruleName": "ApexCRUDViolation",
        "category": "Security",
        "url": "https://pmd.github.io/pmd-6.55.0/pmd_rules_apex_security.html#apexcrudviolation",
        "message": "\nValidate CRUD permission before SOQL/DML operation or enforce user mode\n"
      }
    ]
  }
]
```

**Expected behavior**
We'd expect to only see relative paths in the reports. We'd like to be able to either provide the `--relativize-paths-with` argument or equivalent mechanism where the scanner could itself figure out the root path and pass it down to PMD.

**Desktop (please complete the following information):**
 - OS: macOS 
 - Scanner Version @salesforce/sfdx-scanner 3.16.0

**Additional context**

"Workaround": N/A

"Urgency": Kind of "Business stopping". We want to integrate the scanner as part of our workflows but since we have a ton of code already (which has passed security review a few times) we'd like to focus only on new/modified code and we rely on an external mechanism to create and compare reports from a "baseline".

**Comment by johnbelosf** [comment]
Thank you @bh-sergio-clara we'll discuss internally

**Comment by jfeingold35** [comment]
@rmathewsbeyond , we're in an alpha release for our next gen Code Analyzer v5. One of the improvements in v5 is truncation of paths in outfiles. We encourage you to try it out, and let us know if it meets your needs. https://developer.salesforce.com/docs/platform/salesforce-code-analyzer/guide/version-5.html

**Comment by stephen-carter-at-sf** [comment]
Yes all output formats for Code Analyzer v5 now use relative format instead of absolute where possible. Thus closing this issue as we encourage our users to start migrating to v5:
https://developer.salesforce.com/docs/platform/salesforce-code-analyzer/guide/code-analyzer.html
